addons/mail/models/mail_activity.py

```diff
@@ -86,18 +84,18 @@ def default_get(self, fields):
     default_note = fields.Html(string="Default Note", translate=True)
 
     #Fields for display purpose only
-    initial_res_model_id = fields.Many2one('ir.model', 'Initial model', compute="_compute_initial_res_model_id", store=False,
+    initial_res_model = fields.Selection(selection=_get_model_selection, string='Initial model', compute="_compute_initial_res_model", store=False,
             help='Technical field to keep track of the model at the start of editing to support UX related behaviour')
     res_model_change = fields.Boolean(string="Model has change", help="Technical field for UX related behaviour", default=False, store=False)
 
-    @api.onchange('res_model_id')
-    def _onchange_res_model_id(self):
-        self.mail_template_ids = self.mail_template_ids.filtered(lambda template: template.model_id == self.res_model_id)
-        self.res_model_change = self.initial_res_model_id and self.initial_res_model_id != self.res_model_id
+    @api.onchange('res_model')
+    def _onchange_res_model(self):
+        self.mail_template_ids = self.mail_template_ids.filtered(lambda template: template.sudo().model_id.model == self.res_model)
```

Comment:
Shouldn't we put the sudo directly on self to avoid doing a sub-optimized sudo ?